### PR TITLE
add lookAtUsingOrtho to ViewState3d

### PR DESCRIPTION
**Pull request by MarcNeely**
Add method to set view state with ortho camera definition as defined in BCF.

### CODE REVIEW

**Review by kabentley — DISMISSED** [review]

**Comment by pmconne** [comment]
@MarcNeely if this is ready for review please click "Ready for review" button.

File: core/frontend/src/ViewState.ts [hunk 1]
```diff
@@ -1533,13 +1533,29 @@ export abstract class ViewState3d extends ViewState {
    * adjusted when the [[Viewport]] is synchronized from this view.
    */
   public lookAt(eyePoint: XYAndZ, targetPoint: XYAndZ, upVector: Vector3d, newExtents?: XAndY, frontDistance?: number, backDistance?: number, opts?: ViewChangeOptions): ViewStatus {
+    return this._lookAt(eyePoint, upVector, targetPoint, undefined, newExtents, frontDistance, backDistance, opts);
+  }
+
+  private _lookAt(eyePoint: XYAndZ, upVector: Vector3d, targetPoint?: XYAndZ, viewDirection?: Vector3d, newExtents?: XAndY, frontDistance?: number, backDistance?: number, opts?: ViewChangeOptions): ViewStatus {
```
[the remark this answers is not in the document]
Comment (MarcNeely) [reply core/frontend/src/ViewState.ts]: So is this what you wanted, or did you want me to actually make a breaking change to the LookAt api itself?

File: core/frontend/src/ViewState.ts [hunk 2]
```diff
@@ -63,6 +63,49 @@ export interface ModelDisplayTransformProvider {
   getModelDisplayTransform(modelId: Id64String, baseTransform: Transform): Transform;
 }
 
+/** Interface adopted by caller that wants to set up a perspective or orthographic view for a [[ViewState3d]].
```
Comment (pmconne) [inline core/frontend/src/ViewState.ts:66]: `Arguments for [[ViewState3d.lookAt]]`.

File: core/frontend/src/ViewState.ts [hunk 3]
```diff
@@ -63,6 +63,49 @@ export interface ModelDisplayTransformProvider {
...
+  readonly opts?: ViewChangeOptions;
+}
+
+/** Interface adopted by caller that wants to set up a perspective view for a [[ViewState3d]].
+ * @see [[LookAtArgs]]
+ * @see [[ViewState3d.lookAtPerspectiveOrOrtho]].
+ * @beta
+ */
+export interface LookAtPerspectiveArgs extends LookAtArgs {
+  /** The direction in which the view should look. */
```
Comment (pmconne) [inline core/frontend/src/ViewState.ts:89]: The direction is the vector between this point and the eye point.

File: core/frontend/src/ViewState.ts [hunk 4]
```diff
@@ -1615,17 +1658,46 @@ export abstract class ViewState3d extends ViewState {
...
+   * @param args see [[LookAtPerspectiveArgs]] and [[LookAtOrthoArgs]].
+   * @returns A [[ViewStatus]] indicating whether the camera was successfully positioned.
+   * @note If the aspect ratio of viewDelta does not match the aspect ratio of a Viewport into which this view is displayed, it will be
+   * adjusted when the [[Viewport]] is synchronized from this view.
+   * @beta
+   */
+  public lookAtPerspectiveOrOrtho(args: LookAtPerspectiveArgs | LookAtOrthoArgs): ViewStatus {
+    const pArgs = args as LookAtPerspectiveArgs;
+    const isPerpective = undefined !== pArgs.targetPoint;
+
```
Comment (pmconne) [inline core/frontend/src/ViewState.ts:1674]:
```
if (isPerspective && !this.supportsCamera())
  return ViewStatus.NotCameraView;
```

**Review by pmconne — CHANGES_REQUESTED** [review]
Logic looks good. A few nitpicks. Add some tests?

File: core/frontend/src/ViewState.ts [hunk 5]
```diff
@@ -63,6 +63,49 @@ export interface ModelDisplayTransformProvider {
...
+  /** The direction in which the view should look. */
+  readonly targetPoint: XYAndZ;
+  /** The new size (width and height) of the view rectangle. The view rectangle is on the focus plane centered on the targetPoint.
+   * If newExtents is undefined, the existing size is unchanged.
+   */
+  readonly newExtents?: XAndY;
+}
+
+/** Interface adopted by caller that wants to set up an ortho view for a [[ViewState3d]].
+ * @see [[LookAtArgs]]
```
Comment (pmconne) [inline core/frontend/src/ViewState.ts:98]: Don't need to `@see` base types - the documentation generator automatically produces links to them.
`Arguments to [[ViewState3d.lookAtPerspectiveOrOrtho]].`

Comment (MarcNeely) [reply core/frontend/src/ViewState.ts]: removed the @see for base types.

File: core/frontend/src/ViewState.ts (hunk 2 above)
Comment (MarcNeely) [reply core/frontend/src/ViewState.ts:66]: fix misspell.

**Comment by MarcNeely** [comment]
> Logic looks good. A few nitpicks. Add some tests?

Tests added.

**Review by pmconne — APPROVED** [review]

**Review by markschlosseratbentley — APPROVED** [review]

**Comment by MarcNeely** [comment]
@Mergifyio update

**Pull request merged**